contracts/token/LinearVestingWithPenalty.sol

```diff
@@ -457,8 +407,28 @@ contract LinearVestingWithPenalty is Initializable, ERC20Upgradeable, AccessCont
     }
 
     /**
-     * @dev Authorizes upgrades to the contract.
-     * @param newImplementation The address of the new implementation.
-     */
+    * @dev Overrides the internal _update function to transfer vesting schedules
+    * and enforce admin-only access through the onlyAdmin modifier.
+    */
+    function _update(address from, address to, uint256 value) internal virtual override onlyRole(ADMIN_ROLE) {
```

Comment:
just to clarify that this would only work if admin is the holder of the vested tokens or, if admin is trying to transfer somebody else's vested tokens (via `transferFrom`) this would only work if holder has made the `approve`, right?

Reply:
1) vSOPH are non transferable
2) admin can transfer ...